### PR TITLE
Reworks maps and bug fixing

### PR DESCRIPTION
**Pull request by sashacmc**
- Introduced ThreadSafeMap to simplify locking
 - Add logging
 - Fix query registration and selection

### CODE REVIEW

File: src/ZenohUTransport.cpp [hunk 1]
```diff
@@ -18,6 +18,9 @@
 
 #include <stdexcept>
 
+#define LOG_LEVEL 0
```
Comment (sashacmc) [inline src/ZenohUTransport.cpp:21]: This hardcoded level will be removed when we finished tests.
And I am not sure, may be we can move logging to the up-cpp, or use some external logger?

File: include/up-transport-zenoh-cpp/log.h
Comment (gregmedd) [inline include/up-transport-zenoh-cpp/log.h:1]: I would prefer that you just use spdlog if you need a logging facility. If that does not meet your needs, let me know. I would be happy to provide a logging header that does not use macros.

File: src/ZenohUTransport.cpp (hunk 1 above)
Comment (gregmedd) [reply src/ZenohUTransport.cpp:21]: `spdlog` is already included in the build for this library. I recommend using that

**Review by gregmedd — CHANGES_REQUESTED** [review]
I still need to finish reading the transport cpp file. Adding comments now for feedback on logging.

**Review by gregmedd — APPROVED** [review]
The code change looks good to me.

I would recommend revising your commit messages to provide more complete information (i.e. what bugs were fixed). [This page](https://cbea.ms/git-commit/) has some good guidelines and advice on that.

Git's interactive rebase can be used to squash, reorder, and reword commits.

I won't hold up merging this change over commit messages if it's urgent. Just something to keep in mind for the future.

**Comment by sashacmc** [comment]
> The code change looks good to me.
> 
> I would recommend revising your commit messages to provide more complete information (i.e. what bugs were fixed). [This page](https://cbea.ms/git-commit/) has some good guidelines and advice on that.
> 
> Git's interactive rebase can be used to squash, reorder, and reword commits.
> 
> I won't hold up merging this change over commit messages if it's urgent. Just something to keep in mind for the future.

Ok, but I prefer sqash commits and merge full PR as one commit.
And it will be difficult to separate bugfix from maps rework, because maps rework helped to fix it (first commit not really did it)

**Comment by gregmedd** [comment]
> > The code change looks good to me.
> 
> > 
> 
> > I would recommend revising your commit messages to provide more complete information (i.e. what bugs were fixed). [This page](https://cbea.ms/git-commit/) has some good guidelines and advice on that.
> 
> > 
> 
> > Git's interactive rebase can be used to squash, reorder, and reword commits.
> 
> > 
> 
> > I won't hold up merging this change over commit messages if it's urgent. Just something to keep in mind for the future.
> 
> 
> 
> Ok, but I prefer sqash commits and merge full PR as one commit.
> 
> And it will be difficult to separate bugfix from maps rework, because maps rework helped to fix it (first commit not really did it)

Feel free to squash and force push your commits of you feel they need to be structured differently. We don't use "squash and merge" on these projects in order to allow (and encourage) contributors to 1) provide good commit messages for other users s to read (and help in creating release notes) and 2) curate the commits they submit for review.

If you'd like, I can squash and merge this PR now. However, I generally prefer that the author of the PR be responsible for that. It's the standard I've been holding other contributors to 😅

**Comment by sashacmc** [comment]
> Feel free to squash and force push your commits if you feel they need to be structured differently.

Done!

> We don't use "squash and merge" on these projects in order to allow (and encourage) contributors to 1) provide good commit messages for other users s to read (and help in creating release notes) and 2) curate the commits they submit for review.

I agree, this is good practice.

**Pull request merged**